src/shmem_node_util.h

```diff
@@ -0,0 +1,63 @@
+/* -*- C -*-
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S.  Government
+ * retains certain rights in this software.
+ *
+ * Copyright (c) 2018 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+#ifndef SHMEM_NODE_UTIL_H
+#define SHMEM_NODE_UTIL_H
+
+#include <limits.h>
+#include <sys/param.h>
+
+#ifdef MAXHOSTNAMELEN
+#define SHMEM_INTERNAL_MAX_HOSTNAME_LEN MAXHOSTNAMELEN
+#else
+#define SHMEM_INTERNAL_MAX_HOSTNAME_LEN HOST_NAME_MAX
+#endif
+
+#define SHMEM_INTERNAL_MAX_NPES_PER_NODE 255
```

Comment:
Should this now be `sizeof(int)`?

Reply:
Do these constants need to be exposed in the header, or can the be moved to the .c file?

Reply:
Do you mean `2**sizeof(int)`?  I would hope we could run more than 4 PEs per node... ;)  But then again, I don't think 4294967295 would ever work in practice.

Yes, I think that constant can go in the .c file, but I'm not sure if `MAX_HOSTNAME_LEN` can now that we have the `shmem_transport_addr_t` type which may consist of a string this length.